plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -457,17 +461,53 @@ public static ColumnMapping timestampColumnMapping()
 
     public static ColumnMapping timestampColumnMapping(TimestampType timestampType)
     {
-        checkArgument(timestampType.getPrecision() <= MAX_SHORT_PRECISION, "Precision is out of range: %s", timestampType.getPrecision());
-        return ColumnMapping.longMapping(
+        if (timestampType.getPrecision() <= MAX_SHORT_PRECISION) {
+            return ColumnMapping.longMapping(
+                    timestampType,
+                    timestampReadFunction(timestampType),
+                    timestampWriteFunction(timestampType));
+        }
+        return ColumnMapping.objectMapping(
                 timestampType,
-                timestampReadFunction(timestampType),
-                timestampWriteFunction(timestampType));
+                longTimestampReadFunction(timestampType),
+                longTimestampWriteFunction(timestampType));
     }
 
     public static LongReadFunction timestampReadFunction(TimestampType timestampType)
     {
         checkArgument(timestampType.getPrecision() <= MAX_SHORT_PRECISION, "Precision is out of range: %s", timestampType.getPrecision());
-        return (resultSet, columnIndex) -> toPrestoTimestamp(timestampType, resultSet.getObject(columnIndex, LocalDateTime.class));
+        return (resultSet, columnIndex) -> {
+            long epochMicros = toPrestoTimestamp(timestampType, resultSet.getObject(columnIndex, LocalDateTime.class));
+            verifyTimestamp(epochMicros, 0, timestampType.getPrecision());
+            return epochMicros;
+        };
+    }
+
+    private static ObjectReadFunction longTimestampReadFunction(TimestampType timestampType)
+    {
+        checkArgument(timestampType.getPrecision() > MAX_SHORT_PRECISION, "Precision is out of range: %s", timestampType.getPrecision());
```

Comment:
add
```
checkArgument(timestampType.getPrecision() <= 9
```

since the function is not useful, as it goes via LDT

Reply:
Done!